test/elixir/test/test_helper.exs

```diff
@@ -259,6 +259,59 @@ defmodule CouchTestCase do
         opts = [pretty: true, width: 20, limit: :infinity, printable_limit: :infinity]
         inspect(resp, opts)
       end
+
+      def restart_cluster() do
+        resp = Couch.get("/_membership")
+        assert resp.status_code == 200
+        nodes = resp.body["all_nodes"]
+
+        nodes_ports =
+          Enum.reduce(nodes, [], fn node, acc ->
+            port = node_to_port(node)
+            [{node, port} | acc]
+          end)
+
+        tasks =
+          Enum.map(nodes_ports, fn {node, port} ->
+            Task.async(fn -> restart_node(node, port) end)
+          end)
+
+        Task.yield_many(tasks, length(nodes) * 5000)
+      end
+
+      def restart_node(node \\ "node1@127.0.0.1") do
+        port = node_to_port(node)
+        restart_node(node, port)
+      end
+
+      defp restart_node(node, port) do
+        url = "http://127.0.0.1:#{port}/_node/#{node}/_restart"
+        resp = Couch.post(url)
+        assert HTTPotion.Response.success?(resp)
+        assert resp.body["ok"]
+        # make sure node went down. we assuming the node can't bounce quick
+        # enough to inroduce a race here
+        retry_until(fn -> !node_is_running(port) end)
+        # wait utill node is back
+        retry_until(fn -> node_is_running(port) end, 500, 10000)
+      end
+
+      defp node_is_running(port) do
+        url = "http://127.0.0.1:#{port}/_up"
+        resp = Couch.get(url)
+
+        case HTTPotion.Response.success?(resp) do
+          true -> resp.status_code >= 200 && resp.status_code < 400
```

Comment:
`resp.status_code in (200 .. 399)`

Reply:
Hm.. This does the same thing, you know. I don't see ranges used for status code assertion anywhere else in existing elixir tests, so are you just expressing your personal preferences?

Reply:
Well, whatever, I think my variant is more readable, but I don't care. Changed